source/views/guides/update-guide.markdown

```diff
@@ -0,0 +1,197 @@
+        .parse::<i32>().expect("Invalid ID");
+    let connection = establish_connection();
+
+    let post = diesel::update(posts.find(id))
+        .set(published.eq(true))
+        .get_result::<Post>(&connection)
+        .expect(&format!("Unable to find post {}", id));
+    println!("Published post {}", post.title);
+}
+```
+
+First, we find the post by the id that we pass into the CLI. Then we call .set on the result where we set the column published to be true. Then we grab the result and pass that to our print line. 
+
+The next example will take you through updating other types of columns besides a bool. 
+
+## Passing multiple values into set
+
+Let's say instead of just setting published to true on your post, you also wanted to update the title to tell the world you've published this post. 
+
+```rust
+fn main() {
+    use diesel_update::schema::posts::dsl::{posts, published, title};
+
+    let id = args().nth(1).expect("publish_post requires a post id")
+        .parse::<i32>().expect("Invalid ID");
+    let connection = establish_connection();
+
+    let post = diesel::update(posts.find(id))
+        .set((published.eq(true), title.eq("I've been updated")))
+        .get_result::<Post>(&connection)
+        .expect(&format!("Unable to find post {}", id));
+    println!("Published post {}", post.title);
+}
+```
+
+As you can see here, we've now passed in multiple values to set and it will change both of the columns. If we even wanted to pass in description too, we could do that.
+
+One note is the first like of the function, you can see that we added title to `use diesel_update::schema::posts::dsl::{posts, published, title};`. We have to explicitly tell Diesel what columns we want to update.
+
+## Updating a Post with structs
+
+Diesel will also let us create custom structs that we can use to update our posts. 
+
+Let's create a new file `src/bin/update_post_with_struct.rs`
+
+And in that file, first we will import some things from Diesel:
+
+```rust
+#![feature(custom_derive, custom_attribute, plugin)]
+#![plugin(diesel_codegen, dotenv_macros)]
+
+extern crate diesel_demo;
+extern crate diesel;
+
+use self::diesel::prelude::*;
+use self::diesel_demo::*;
+use self::diesel_demo::models::Post;
+use self::diesel_demo::schema::posts;
+use std::env::args;
+
+```
+
+Our setup is identical to `publish_post` with 3 exceptions. 
+
+In order to use a struct to update our posts, we have to create a custom changeset for Posts. Adding the first two lines will allow us to do this. The other addition we made was adding `use self::diesel_update::schema::posts;`
+
+Now let's add our struct below the setup.
+
+```rust
+#[changeset_for(posts)]
+struct UpdatePost {
+    id: i32,
+    title: String,
+    body: Option<String>,
+}
+```
+
+Here we're creating a change set for a Post. The struct is called UpdatePost and the attributes we want are id, title, and body. Though here, we are saying that the body can be optional.
```

Comment:
We maybe should expand on `here, we are saying that the body can be optional`. It might be useful to explain that in this app we're building, `title` will always be required, but a body/description will be an optional update, which Diesel supports through using `Option`s of the params you want to conditionally update.